### PR TITLE
Feature/webumenia 1034: 1989 frontpage

### PR DESCRIPTION
**Pull request by martin-havala**
not working newsletter so far [plugin was able to load, got response from subscription, but no email in my mailbox :( ], will check tomorrow

### CODE REVIEW

File: app/Http/Controllers/SubscriptionController.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Newsletter;
+
+class SubscriptionController extends Controller
+{
+    public function subscribe(Request $request)
```
Comment (eronisko) [inline app/Http/Controllers/SubscriptionController.php]: Nit pick, for a more standard route name:
```suggestion
    public function store(Request $request)
```

File: resources/views/welcome.blade.php [hunk 2]
```diff
@@ -1,99 +1,319 @@
...
-                    <a href="https://laravel-news.com">News</a>
-                    <a href="https://blog.laravel.com">Blog</a>
-                    <a href="https://nova.laravel.com">Nova</a>
-                    <a href="https://forge.laravel.com">Forge</a>
-                    <a href="https://github.com/laravel/laravel">GitHub</a>
-                </div>
+    }
+
+    body {
+        background: url('../images/intro/BG-0{{random_int(2, 5)}}.jpg') fixed;
```
Comment (eronisko) [inline resources/views/welcome.blade.php]: This should probably use the [`asset`](https://laravel.com/docs/master/helpers#method-asset) helper too.

File: resources/sass/app.scss [hunk 3]
```diff
@@ -1,8 +1,48 @@
-// Fonts
-@import url('https://fonts.googleapis.com/css?family=Nunito');
+//fonts
+@import '../fonts/gabriello.css';
+@import '../fonts/avenir.css';
```
Comment (eronisko) [inline resources/sass/app.scss]: It would be good if this compiled even if the font file does not exist.

File: .gitignore [hunk 4]
```diff
@@ -14,4 +14,5 @@
 Homestead.json
 Homestead.yaml
 npm-debug.log
-yarn-error.log
\ No newline at end of file
+yarn-error.log
+resources/fonts
```
Comment (igor-kamil) [inline .gitignore]: there should be `.gitignore` file inside `resources/fonts/` 
with
```
*
!.gitignore
```
we want to keep that folder inside the version control - but just not it's contents

File: resources/sass/app.scss [hunk 5]
```diff
@@ -1,8 +1,48 @@
...
 
 // Variables
 @import 'variables';
 
 // Bootstrap
 @import '~bootstrap/scss/bootstrap';
+
+html,
+body {
+    font-family: 'AvenirLTPro-Roman';
```
Comment (igor-kamil) [inline resources/sass/app.scss]: better to use bootstrap variables
```
$font-family-base:           'AvenirLTPro-Roman';
```

File: resources/sass/app.scss [hunk 6]
```diff
@@ -1,8 +1,48 @@
...
+    font-size: 11pt;
+}
+
+h1,
+h2,
+h3,
+h4 {
+    font-family: 'Gabriello Web';
```
Comment (igor-kamil) [inline resources/sass/app.scss]: same here. please use bootstrap variables.
copy these you want to change from `node_modules/bootstrap/scss/_variables.scss` and store them in `resources/sass/_variables.scss`

File: resources/views/welcome.blade.php [hunk 7]
```diff
@@ -1,99 +1,319 @@
...
+
+
+$("#form").submit(function($frm){
+
+    $.ajaxSetup({
+        headers: {
+            'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
+        }
+    });
+    console.log($frm);
```
Comment (igor-kamil) [inline resources/views/welcome.blade.php]: leftover

File: routes/api.php [hunk 8]
```diff
@@ -17,4 +18,9 @@
     return $request->user();
 });
 
+Route::get('/day/{date}', function ($date) {
+    return JournalEntry::where('written_at', '=', $date)->get();
+})->name('api.day');
+
```
Comment (eronisko) [inline routes/api.php]: Probably a merge artifact? This should be covered by line 26. I'd suggest removing this.

Comment (martin-havala) [reply routes/api.php]: sure

File: resources/views/welcome.blade.php [hunk 9]
```diff
@@ -200,12 +220,29 @@
...
     }
 
+    #overlay {
+        position: fixed;
+        top: 0;
+        bottom: 0;
+        left: 0;
+        right: 0;
+        z-index: 0;
+        background: rgba(0, 0, 0, .6);
```
Comment (igor-kamil) [inline resources/views/welcome.blade.php]: it's bit too dark. I think it can be smaller then `0.6`... imho.

**Pull request merged**